libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/NumberInput.cs

```diff
@@ -109,17 +108,8 @@ protected override Task<InputState> OnRecognizeInputAsync(DialogContext dc, Canc
 
         private string GetCulture(DialogContext dc)
         {
-            if (!string.IsNullOrEmpty(dc.Context.Activity.Locale))
-            {
-                return dc.Context.Activity.Locale;
-            }
-
-            if (this.DefaultLocale != null)
-            {
-                return this.DefaultLocale.GetValue(dc.State);
-            }
-
-            return English;
+            // Note: Default locale will be considered for deprecation as part of 4.13.
+            return dc.GetLocale() ?? DefaultLocale.GetValue(dc.State) ?? string.Empty;
```

Comment:
DefaultLocale?.GetValue(dc.State)

Reply:
Good catch! The code after dc.GetLocale() will never really be executed but until we officially deprecate this DefaultLocale good to have the code be correct.